include/i2c.h

```diff
@@ -395,13 +420,49 @@ static inline int i2c_read(struct device *dev, u8_t *buf,
 	return i2c_transfer(dev, &msg, 1, addr);
 }
 
+/**
+ * @brief Write then read data from an I2C device.
+ *
+ * This supports the common operation "this is what I want", "now give
+ * it to me" transaction pair through a combined write-then-read bus
+ * transaction.
+ *
+ * @param dev Pointer to the device structure for the driver instance
+ * @param write_buf Pointer to the data to be written
+ * @param num_write Number of bytes to write
+ * @param read_buf Pointer to storage for read data
+ * @param num_read Number of bytes to read
+ * @param addr Address of the I2C device
+ *
+ * @retval 0 if successful
+ * @retval negative on error.
+ */
+static inline int i2c_write_read(struct device *dev,
+				 const void *write_buf, size_t num_write,
+				 void *read_buf, size_t num_read,
+				 u16_t addr)
```

Comment:
There is one more issue regarding `i2c_write_read` function which we've overlooked. I2C functions need to support 10-bit addressing. At the moment the API supports this using two defines: 
- `I2C_ADDR_10_BITS` passed to `i2c_configure`. This is deprecated
- `I2C_MSG_ADDR_10_BITS` passed as part of `struct i2c_msg` to `i2c_transfer` function.

According to the I2C standard addressing mode can be set on a per message basis. As such `i2c_write_read` also needs to support it. I see two possibilities:
- add an extra flags/cfg parameter
- reuse `u16_t addr` parameter and have user set 10-bit mode by doing something like `I2C_MSG_ADDR_10_BIT | addr` when calling the function. `I2C_MSG_ADDR_10_BIT` would need to be redefined as `1 << 15`.

The former approach seems somehow cleaner and would allow to extend the function in the future (if ever required).

Reply:
You have a point, but none of the existing I2C convenience functions including `i2c_write` and `i2c_read` support 10-bit addresses.  So this is a new feature that I believe is out of scope for this PR, and you should open an issue for it.

Reply:
>  So this is a new feature that I believe is out of scope for this PR

As this is a new feature we definitely need to consider all the requirements. Implementing first approach "add an extra flags/cfg parameter" at a later stage would force us to go through another deprecation cycle. On the other hand, if we decide that the second option "reuse `u16_t addr` parameter" is acceptable we don't need to add 10-bit addressing now.